src/Workspaces/Core/Portable/LanguageServices/SyntaxFactsService/ISyntaxFactsServiceExtensions.cs

```diff
@@ -41,10 +39,46 @@ public static bool IsRegularOrDocumentationComment(this ISyntaxFactsService synt
             syntaxFacts.GetPartsOfAssignmentStatement(statement, out left, out _, out right);
         }
 
+        public static SyntaxNode Unparenthesize(
+            this ISyntaxFactsService syntaxFacts, SyntaxNode node)
+        {
+            syntaxFacts.GetPartsOfParenthesizedExpression(node,
+                out var openParenToken, out var expression, out var closeParenToken);
+
+            var leadingTrivia = openParenToken.LeadingTrivia
+                .Concat(openParenToken.TrailingTrivia)
+                .Where(t => !syntaxFacts.IsElastic(t))
+                .Concat(expression.GetLeadingTrivia());
+
+            var trailingTrivia = expression.GetTrailingTrivia()
+                .Concat(closeParenToken.LeadingTrivia)
+                .Where(t => !syntaxFacts.IsElastic(t))
+                .Concat(closeParenToken.TrailingTrivia);
+
+            var resultNode = expression
+                .WithLeadingTrivia(leadingTrivia)
+                .WithTrailingTrivia(trailingTrivia);
+
+            return resultNode;
+        }
+
         public static bool SpansPreprocessorDirective(this ISyntaxFactsService service, SyntaxNode node)
             => service.SpansPreprocessorDirective(SpecializedCollections.SingletonEnumerable(node));
 
         public static bool IsWhitespaceOrEndOfLineTrivia(this ISyntaxFactsService syntaxFacts, SyntaxTrivia trivia)
             => syntaxFacts.IsWhitespaceTrivia(trivia) || syntaxFacts.IsEndOfLineTrivia(trivia);
+
+        public static void GetPartsOfBinaryExpression(
```

Comment:
❓ I don't see where `GetPartsOfBinaryExpression` and `GetExpressionOfParenthesizedExpression` are used here. Were they supposed to be in a different commit?

:bulb: This method should be named `GetExpressionsOfBinaryExpression` for consistency with the next method. It might further make sense to return `(SyntaxNode left, SyntaxNode right)` instead of using `out` variables.

Reply:
The naming consistency is either 'Parts' to indicate the deconstruction when there are multiple things returned, or 'GetXXXOfYYY' where 'XXX' is the name of the actual child property if only one child is being returned.  This is consistent with the rest of hte naming in this file.

Reply:
Note: now that we have tuples, these "GetPartsOfXXX" members will likely become DeconstructXXX methods in teh future.